sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1371,6 +1412,8 @@ struct _pi_event : _pi_object {
 
   bool hasExternalRefs() { return RefCountExternal != 0; }
 
+  bool IsDiscarded = false;
```

Comment:
This is missing a comment, but also isn't it possible to know that from *external* ref-count being 0?

Reply:
I will add a comment. IsDiscarded means that pointer to the event is not visible externally at all, from the moment creation (i.e. event was explicitly discarded). !RefCountExternal() == true means event may have been visible at the moment of creation but then became invisible at some point.
I have separated these two things because using optimization only for explicitly discarded events (have IsDiscarded = true)  is the safest and robust way and I wanted to create a switch where we can limit the optimization only for events which were explicitly discarded.

Reply:
Please add some of the explanation to the comment, and maybe some TODO path forward of when/how this can be removed.

Reply:
Added command and TODO.